src/meep.hpp

```diff
@@ -386,24 +386,26 @@ class h5file {
 
   bool ok();
 
-  realnum *read(const char *dataname, int *rank, size_t *dims, int maxrank);
-  void write(const char *dataname, int rank, const size_t *dims, realnum *data,
-             bool single_precision = true);
```

Comment:
I think we should keep `single_precision` as the default in general, just to keep file sizes small by default.   Anything that is dumping and restoring data should pass the working precision, however.

Reply:
Done.